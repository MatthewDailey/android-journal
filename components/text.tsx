import { StyleSheet, Text, TextProps, TextInput, TextInputProps } from "react-native";

const fontSize = 16
const textStyle = StyleSheet.create({
    normal: {
        fontWeight: "normal",
        fontSize,
    },
    bold: { 
        fontWeight: "bold",
        fontSize,
    },
    light: {
        fontWeight: "normal",
        fontSize: 12,
        opacity: 0.5,
    },
    normalInput: {
        fontWeight: "normal",
        fontSize,
        borderWidth: 0,
    }
});

export const HeavyText = (props: TextProps) => (
    <Text style={textStyle.bold} {...props}/>
)

export const NormalText = (props: TextProps) => (
    <Text style={textStyle.normal} {...props}/>
)

export const LightText = (props: TextProps) => (
    <Text style={textStyle.light} {...props}/>
)

export const NormalTextInput = (props: TextInputProps) => (
    <TextInput style={textStyle.normalInput} {...props}/>
)