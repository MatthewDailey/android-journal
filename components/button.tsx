import React from "react";
import { StyleSheet, View, Button, ButtonProps, Pressable } from "react-native";
import { HeavyText } from "./text";

const spacing = 16;
const styles = StyleSheet.create({
    button: {
        color: "#000",
        padding: spacing,
        borderRadius: 8,
        borderColor: "#000",
        borderWidth: 2,
        fontSize: 16,
        flexGrow: 1,
        alignItems: "center",
    },
    buttonContainer: {
        borderTopColor: "#000",
        borderTopWidth: 2,
        padding: spacing,
        display: "flex",
        flexDirection: "row",
    }
})

export const PrimaryButton = (props: { text: string, onPress: () => void }) => (
    <Pressable 
        style={({pressed}) => {
            return {
                backgroundColor: pressed ? "#ffeedb" : "#fff",
                ...styles.button
            }
        }} 
        onPress={props.onPress}>
        <HeavyText>{props.text}</HeavyText>
    </Pressable>
)

export const ButtonContainer = (props: { children: React.ReactNode[]}) => {
    const views = [];

    for (let i = 0; i < props.children.length; i++) {
        views.push(props.children[i]);
        if (i < props.children.length - 1) {
            views.push(<View style={{width: spacing / 2}} key={i}/>);
        }
    }

    return (
        <View style={styles.buttonContainer}>
            {views}
        </View>
    )
}