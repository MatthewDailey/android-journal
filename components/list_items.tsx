import React from "react";
import { StyleSheet, Text, View } from "react-native";
import { LightText, NormalText } from "./text";

const styles = StyleSheet.create({
    gratitude: {},
});

export const GratitudeListItem = (props: { text: string, dateMs: number }) => {
    const date = new Date(props.dateMs);
    const hoursAndMinutesString = `${date.getHours()%12}:${date.getMinutes()}${date.getHours() > 12 ? "pm" : "am"}`; 
    return (
        <View style={styles.gratitude}>
            <LightText>{hoursAndMinutesString} â€¢ I'm grateful for...</LightText>
            <NormalText>{props.text}</NormalText>
        </View>
    )
}