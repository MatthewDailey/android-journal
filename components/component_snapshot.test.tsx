import React from "react"
import renderer from 'react-test-renderer'
import { ButtonContainer, PrimaryButton } from "./button"
import { DateListItem, EditingListItem, GratitudeListItem, JournalListItem, NoEntryListItem } from "./list_items"
import { HeavyText, LightText, NormalText, NormalTextInput } from "./text"

describe('Component Snapshot Tests', () => {
    it('PrimaryButton', () => {    
        const tree: any = renderer.create(<PrimaryButton onPress={() => { } } text="Journal" />).toJSON()
        expect(tree).toMatchSnapshot()
    })
    it('ButtonContainer', () => {    
        const tree: any = renderer.create(
            <ButtonContainer>
                <PrimaryButton onPress={() => { } } text="Journal" />
                <PrimaryButton onPress={() => { } } text="Gratitude" />
            </ButtonContainer>
        ).toJSON()
        expect(tree).toMatchSnapshot()
    })
    it('HeavyText', () => {
        const tree: any = renderer.create(<HeavyText>October 5, 2022</HeavyText>).toJSON()
        expect(tree).toMatchSnapshot()
    })
    it('LightText', () => {
        const tree: any = renderer.create(<LightText>5:04pm</LightText>).toJSON()
        expect(tree).toMatchSnapshot()
    })
    it('NormalText', () => {
        const tree: any = renderer.create(<NormalText>Exerted enterence focus hear him</NormalText>).toJSON()
        expect(tree).toMatchSnapshot()
    })
    it('NormalTextInput', () => {
        const tree = renderer.create(<NormalTextInput />).toJSON()
        expect(tree).toMatchSnapshot()
    })
    it('EditingListItem', () => {
        const tree = renderer.create(<EditingListItem header="I'm grateful for..." onChangeText={function (text: string): void {} } />).toJSON()
        expect(tree).toMatchSnapshot()
    })
    it('GratitudeListItem', () => {
        const tree = renderer.create(<GratitudeListItem text="This is a long string that will wrap to many lines that needs to be truncated so that this element does not take up too much space." dateMs={1665553993440} />).toJSON()
        expect(tree).toMatchSnapshot()
    })
    it('JournalListItem', () => {
        const tree = renderer.create(<JournalListItem text="This is a long string that will wrap to many lines that needs to be truncated so that this element does not take up too much space." dateMs={1665553993440} />).toJSON()
        expect(tree).toMatchSnapshot()
    })
    it('DateListItem', () => {
        const tree = renderer.create(<DateListItem dateMs={new Date('2022-10-10T20:12:30.913Z').getTime()}/>).toJSON()
        expect(tree).toMatchSnapshot()
    })
    it('NoEntryListItem', () => {
        const tree = renderer.create(<NoEntryListItem />).toJSON()
        expect(tree).toMatchSnapshot()
    })
})