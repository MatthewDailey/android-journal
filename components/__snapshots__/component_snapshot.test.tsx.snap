// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Snapshot Tests ButtonContainer 1`] = `
<View
  style={
    Object {
      "borderTopColor": "#000",
      "borderTopWidth": 2,
      "display": "flex",
      "flexDirection": "row",
      "padding": 16,
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderColor": "#000",
        "borderRadius": 8,
        "borderWidth": 2,
        "color": "#000",
        "flexGrow": 1,
        "fontSize": 16,
        "padding": 16,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      Journal
    </Text>
  </View>
  <View
    style={
      Object {
        "width": 8,
      }
    }
  />
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderColor": "#000",
        "borderRadius": 8,
        "borderWidth": 2,
        "color": "#000",
        "flexGrow": 1,
        "fontSize": 16,
        "padding": 16,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      Gratitude
    </Text>
  </View>
</View>
`;

exports[`Component Snapshot Tests DateListItem 1`] = `
<View
  style={
    Object {
      "padding": 8,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 16,
        "fontWeight": "bold",
      }
    }
  >
    October 10, 2022
  </Text>
</View>
`;

exports[`Component Snapshot Tests EditingListItem 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFEEDB",
      "paddingBottom": 8,
      "paddingLeft": 16,
      "paddingRight": 16,
      "paddingTop": 8,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 12,
        "fontWeight": "normal",
        "opacity": 0.5,
      }
    }
  >
    I'm grateful for...
  </Text>
  <TextInput
    autoFocus={true}
    multiline={true}
    onChangeText={[Function]}
    style={
      Object {
        "borderWidth": 0,
        "fontSize": 16,
        "fontWeight": "normal",
      }
    }
    testID="edit_input"
  />
</View>
`;

exports[`Component Snapshot Tests GratitudeListItem 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "paddingBottom": 8,
      "paddingLeft": 16,
      "paddingRight": 16,
      "paddingTop": 8,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 12,
        "fontWeight": "normal",
        "opacity": 0.5,
      }
    }
  >
    10:53pm â€¢ I'm grateful for...
  </Text>
  <Text
    numberOfLines={2}
    style={
      Object {
        "fontSize": 16,
        "fontWeight": "normal",
      }
    }
  >
    This is a long string that will wrap to many lines that needs to be truncated so that this element does not take up too much space.
  </Text>
</View>
`;

exports[`Component Snapshot Tests HeavyText 1`] = `
<Text
  style={
    Object {
      "fontSize": 16,
      "fontWeight": "bold",
    }
  }
>
  October 5, 2022
</Text>
`;

exports[`Component Snapshot Tests JournalListItem 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "paddingBottom": 8,
      "paddingLeft": 16,
      "paddingRight": 16,
      "paddingTop": 8,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 12,
        "fontWeight": "normal",
        "opacity": 0.5,
      }
    }
  >
    10:53pm
  </Text>
  <Text
    numberOfLines={2}
    style={
      Object {
        "fontSize": 16,
        "fontWeight": "normal",
      }
    }
  >
    This is a long string that will wrap to many lines that needs to be truncated so that this element does not take up too much space.
  </Text>
</View>
`;

exports[`Component Snapshot Tests LightText 1`] = `
<Text
  style={
    Object {
      "fontSize": 12,
      "fontWeight": "normal",
      "opacity": 0.5,
    }
  }
>
  5:04pm
</Text>
`;

exports[`Component Snapshot Tests NoEntryListItem 1`] = `
<View
  style={
    Object {
      "paddingBottom": 8,
      "paddingLeft": 16,
      "paddingRight": 16,
      "paddingTop": 8,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 16,
        "fontWeight": "normal",
      }
    }
  >
    NO ENTRY TODAY
  </Text>
</View>
`;

exports[`Component Snapshot Tests NormalText 1`] = `
<Text
  style={
    Object {
      "fontSize": 16,
      "fontWeight": "normal",
    }
  }
>
  Exerted enterence focus hear him
</Text>
`;

exports[`Component Snapshot Tests NormalTextInput 1`] = `
<TextInput
  style={
    Object {
      "borderWidth": 0,
      "fontSize": 16,
      "fontWeight": "normal",
    }
  }
/>
`;

exports[`Component Snapshot Tests PrimaryButton 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "borderColor": "#000",
      "borderRadius": 8,
      "borderWidth": 2,
      "color": "#000",
      "flexGrow": 1,
      "fontSize": 16,
      "padding": 16,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 16,
        "fontWeight": "bold",
      }
    }
  >
    Journal
  </Text>
</View>
`;
